integrations/clickup/main.py

```diff
@@ -0,0 +1,140 @@
+from enum import StrEnum
+from typing import Any, Dict
+from loguru import logger
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import ASYNC_GENERATOR_RESYNC_TYPE
+from client import ClickupClient
+
+EVENT_ACTIONS = {
+    "listCreated": "register",
+    "listUpdated": "register",
+    "listDeleted": "unregister",
+    "taskCreated": "register",
+    "taskUpdated": "register",
+    "taskDeleted": "unregister",
+}
+
+
+class ObjectKind(StrEnum):
+    TEAM = "team"
+    PROJECT = "project"
+    ISSUE = "issue"
+
+
+async def init_client() -> ClickupClient:
+    client = ClickupClient(
+        ocean.integration_config["clickup_base_url"],
+        ocean.integration_config["clickup_personal_token"],
+        ocean.integration_config["clickup_archived_parameter"],
+    )
+    return client
+
+
+async def setup_application() -> None:
+    app_host = ocean.integration_config.get("app_host")
+    if not app_host:
+        logger.warning(
+            "App host was not provided, skipping webhook creation. "
+            "Without setting up the webhook, you will have to manually initiate resync to get updates from ClickUp."
+        )
+        return
+    clickup_client = await init_client()  # Await the initialization of the client
+    await clickup_client.create_clickup_events_webhook(app_host)
+
+
+@ocean.on_resync(ObjectKind.TEAM)
+async def on_resync_teams(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    clickup_client = await init_client()
+    async for teams in clickup_client.get_clickup_teams():
+        logger.info(f"Received team of length {len(teams)}")
+        yield teams
+
+
+@ocean.on_resync(ObjectKind.PROJECT)
+async def on_resync_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    clickup_client = await init_client()
+    async for projects in clickup_client.get_folderless_projects():
+        logger.info(f"Received project batch with {len(projects)} projects")
+        yield projects
+
+
+@ocean.on_resync(ObjectKind.PROJECT)
+async def on_resync_folder_projects(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    clickup_client = await init_client()
+    async for projects in clickup_client.get_folder_projects():
+        logger.info(f"Received project batch with {len(projects)} projects")
+        yield projects
+
```

Comment:
Have you tested the behaviour of these resync events from the Data Sources page to understand what is happening since they both refer to the same objectkind?

Reply:
Yes, and I have combined them